refactor: use platform/runtime plugin components

edoardo opened the pull request:
Fixes [DHIS2-17268](https://dhis2.atlassian.net/browse/DHIS2-17268)

**Relates to https://github.com/dhis2/data-visualizer-app/pull/3082**
**Relates to https://github.com/dhis2/line-listing-app/pull/396**
**Relates to https://github.com/dhis2/maps-app/pull/3232**

---

### Key features

1. use plugin components from `app-runtime`
2. fix Cypress login for >=41

---

### Description

Generic components for plugins are available in `app-platform` and `app-runtime`.
This PR is removing the original iframe plugin with the message handling that has been "moved" to the platform.
All analytics apps have a correspondent PR for using the platform plugin build with embedded message handling.

Cypress started to fail in >=41 because the login mechanism changed.
Dashboard is still using v6.9.1 of Cypress which prevents us from upgrading `cli-utils-cypress` to a recent version with support for the new login.
The temporary solution is to add a custom login command which accommodates login for all DHIS2 versions.

---

### TODO

-   [x] figure out how to transmit the `removeCachedData` message to the plugin without re-rendering
-   [ ] handle `recordOnNextLoad` to avoid multiple recordings
-   [ ] test plugin type switch (ie. DV to Maps)
-   [ ] test installation status stuff

---

### Known issues

-   [ ] an infinite loop happened and [this](https://github.com/dhis2/dashboard-app/commit/03059501e58e2940efa5a750dac099cb612e5c29) seems to solve it, I'm not sure why and if this is a proper solution.

---

### Screenshots
<img width="704" alt="Screenshot 2023-06-21 at 16 25 33" src="https://github.com/dhis2/dashboard-app/assets/150978/25d8f64f-3a86-4636-9dc8-12148d498444">





[DHIS2-17268]: https://dhis2.atlassian.net/browse/DHIS2-17268?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

edoardo committed:
chore: use alpha versions of platform and runtime

edoardo committed:
refactor: use generic Plugin component from the platform

edoardo committed:
chore: update app-runtime, cli-app-scripts deps

edoardo committed:
refactor: handle recording on first load, pass width and height

edoardo committed:
fix: regenerate yarn.lock file

edoardo committed:
chore: update app-runtime and cli-app-scripts to latest

edoardo committed:
fix: replace deprecated prop with new one

This silences warnings in the console.

edoardo committed:
fix: avoid warning about nested a tags

Link is rendering an a tag, and MenuItem does the same.
Use a redirect on click instead (already done in the print submenu)

edoardo committed:
fix: silence linter warning

This hopefully has a better solution.

edoardo committed:
refactor: simplify communication for offline caching

It should be enough to just use isParentCached for knowing when to start
recording and removing the cache in the plugins.

edoardo committed:
chore: bump cli-app-scripts dependency

edoardo committed:
chore: bump cli-app-scripts

This should fix some issues with plugin content not being visible.

edoardo committed:
chore: remove unused post-robot dependency

edoardo committed:
chore: fix duplicate app-runtime

edoardo committed:
chore: resolve i18next to <21 to avoid plural strings issues

See https://dhis2.atlassian.net/browse/DHIS2-16591.

edoardo committed:
chore: remove i18next direct dep, resolve i18next-scanner

edoardo committed:
chore: cleanup deps to fix failing build

edoardo committed:
chore: add missing cli-style dep, remove ui resolution

edoardo committed:
fix: fix linting error

edoardo committed:
chore: run prettier to fix linting errors

edoardo committed:
test: update snapshots

edoardo committed:
fix: move visualization fetch in try/catch

This is to avoid the error boundary to show up for uncaught errors
(failed to fetch a visualization) which
interferes with Cypress testing.

jenniferarnesen commented on src/pages/edit/ItemSelector/__tests__/__snapshots__/ContentMenuItem.spec.js.snap:
```diff
@@ -3,14 +3,17 @@
 <div>
   <li
-    class="jsx-1930534478 "
+    class="jsx-3192368000 "
```
Maybe this test needs to be changed so we don't get these diffs when the ui version is upgraded.

edoardo committed:
chore: update cli-app-scripts, remove resolution for i18next-scanner

edoardo committed:
chore: restore lost deps

edoardo committed:
Merge remote-tracking branch 'origin/master' into refactor/use-platform-plugin-components

edoardo committed:
fix: extend compatible version check to DV/Maps

This is needed as the new plugin system is used, because the messages
used in the communication have changed.

edoardo committed:
chore: regenerate pot file

edoardo committed:
Merge remote-tracking branch 'origin/master' into refactor/use-platform-plugin-components

edoardo committed:
test: fix test for version compatibility

edoardo committed:
chore: regenerate yarn.lock

edoardo committed:
refactor: move error screen to component for easier reuse

jenniferarnesen commented:
Superceded by https://github.com/dhis2/dashboard-app/pull/3184
